packages/typespec-client-generator-core/src/methods.ts

```diff
@@ -517,7 +517,12 @@ function getServiceMethodLroMetadata(
     finalStateVia: rawMetadata.finalStateVia,
     finalResponse: getFinalResponse(),
     finalStep:
-      rawMetadata.finalStep !== undefined ? { kind: rawMetadata.finalStep.kind } : undefined,
+      rawMetadata.finalStep !== undefined ? { 
+        responseModel: diagnostics.pipe(
+          getClientTypeWithDiagnostics(context, rawMetadata.finalStep.responseModel),
+        ) as SdkModelType,
+        kind: rawMetadata.finalStep.kind 
+      } : undefined,
     pollingStep: {
```

Comment:
@iscai-msft just a heads up, I think polling step might have a lot of the same metadata returned on it